### PR TITLE
Add unit tests for sposet input styles

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Step 1 of #2070
Introduced unit tests for current and desired sposet input styles. Some within determinants. Those styles with sposet outside determinantset are the only ones we are going to support in the future.

Next PRs will settle
1. make LCAO h5 basis initialization controlled by input and respect name tag. RMG needs two basissets for different spins.
2. Add deprecation warning to all deprecated styles. This will go to the next release and then we can delete those deprecated styles in the release after next.

## What type(s) of changes does this code introduce?
- Bugfix
- Unit tests

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
Bora

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by ye-luo** [comment]
Ready for review.

**Comment by prckent** [comment]
I tweaked the error message which has needed improving for some time. However, would it ever be advisable to not abort in this scenario?

**Comment by ye-luo** [comment]
> I tweaked the error message which has needed improving for some time. However, would it ever be advisable to not abort in this scenario?

This part of code was touched because of formatting. It was added for catching badbit caused by de-normal number and I think we should throw an exception. Will change it in a separate PR after more testing.

File: src/OhmmsData/libxmldefs.h [hunk 1]
```diff
@@ -128,8 +128,8 @@ inline bool putContent(std::vector<T>& a, const xmlNodePtr cur)
       b.push_back(t);
     else if (!stream.eof() && stream.fail())
     {
-      std::cerr << "failbit detected when reading type (type_info::name) "
-                << typeid(T).name() << ", value " << t << std::endl;
+      std::cerr << "Error parsing XML for type (type_info::name) " << typeid(T).name() << ", value " << t
```
Comment (PDoakORNL) [inline src/OhmmsData/libxmldefs.h:131]: I assume this is the error which is going to become an exception?

File: src/QMCWaveFunctions/SPOSetBuilderFactory.cpp [hunk 2]
```diff
@@ -301,37 +300,42 @@ void SPOSetBuilderFactory::build_sposet_collection(xmlNodePtr cur)
...
-  app_summary() << "   ------------------------------------------" << std::endl;
+  app_summary() << "   Single particle orbitals (SPO) collection" << std::endl;
+  app_summary() << "   -----------------------------------------" << std::endl;
   app_summary() << "    Name: " << collection_name << "   Type input: " << collection_type << std::endl;
   app_summary() << std::endl;
 
   // create the SPOSet builder
   SPOSetBuilder* bb = createSPOSetBuilder(cur);
+
+  // going through a list of basisset entries
```
Comment (PDoakORNL) [inline src/QMCWaveFunctions/SPOSetBuilderFactory.cpp:311]: basis set is two words I think the cname should be "basis_set"

Not addressable here should we start to make a list of input names to update?

**Review by PDoakORNL — APPROVED** [review]

File: src/OhmmsData/libxmldefs.h (hunk 1 above)
Comment (prckent) [reply src/OhmmsData/libxmldefs.h:131]: Yes

File: src/QMCWaveFunctions/SPOSetBuilderFactory.cpp (hunk 2 above)
Comment (ye-luo) [reply src/QMCWaveFunctions/SPOSetBuilderFactory.cpp:311]: I vote for not changing. It is short enough. Long ones need breakup. If change, we need proper protections.

**Pull request merged**